chromadb/utils/embedding_functions.py

```diff
@@ -581,6 +581,105 @@ def __call__(self, input: Union[Documents, Images]) -> Embeddings:
         return embeddings
 
 
+class AmazonBedrockEmbeddingFunction(EmbeddingFunction[Documents]):
+    def __init__(
+        self,
+        profile_name: Optional[str] = None,
+        region: Optional[str] = None,
+        model_name: str = "amazon.titan-embed-text-v1",
```

Comment:
I think there's a cleaner API boundary here and I'd like to hear your thoughts.

This constructor could accept a `boto3.Session` and optional kwargs to pass into the `client()` call. That way we give users full access to the configuration options for both their `Session` and their `client`, and we have less code to maintain (we don't need to read from `os.environ` for example). I may be missing something here -- is there a reason we need to accept these and construct the `Session` ourselves? If not could we accept a `Session` and config kwargs?

Reply:
I'm okay to make just passing a `boto3.Session` instead of creating session within the class.

I don't have a specific reason to create a `boto3.Session` inside the class. I tried to align with the existing codes. Looking at the other classes creates a client internally, and `HuggingFaceEmbeddingFunction` creates `requests.Session` within the class.

Reply:
btw, do you think we should support all the possible config kwargs as much as possible?

Looking at [the boto3 document](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html#boto3.session.Session.client), I think most of the kwargs are covered by the session, so we may need to pass `api_version`, `use_ssl`, `verify`, and `endpoint_url`, at least.

Reply:
Tentatively, I removed most of the redundant arguments without adding client-specific arguments. Let me know your thoughts on it.
[eedf49d](https://github.com/chroma-core/chroma/pull/1361/commits/eedf49d1717b6177563a15a282986236d2681908)

Reply:
This is already looking way better -- thank you!

One more small ask and then we can check it in: could we have the constructor accept `**kwargs` and pass them directly to the `client()` we create? That would give us three wins:
1 - We don't have to manage the `retry_config` which I imagine some folks will want control over.
2 - Users can specify whatever they want to override in their `Session` config without us needing to make any code changes.
3 - We don't need to import `boto3` in this codebase anymore.

Reply:
Fair point. Fixed by using `**kwargs` a53e415f8c311cf30ba2b541dc28444f6ed185a3